### ISSUE TITLE
Error: Problem with `mutate()` input `.is_null`. - New Factor Levels

### ISSUE THREAD

**Issue by vmandela99** [business-science/modeltime#46]
While checking the accuracy table on the holdout test using the function modeltime_accuracy() throws me an error. Help!

``` r
library(modeltime)
library(tidymodels)
#> -- Attaching packages ---------------- tidymodels 0.1.0 --
#> v broom     0.7.0      v recipes   0.1.13
#> v dials     0.0.8      v rsample   0.0.7 
#> v dplyr     1.0.2      v tibble    3.0.3 
#> v ggplot2   3.3.2      v tune      0.1.1 
#> v infer     0.5.3      v workflows 0.2.1 
#> v parsnip   0.1.3      v yardstick 0.0.6 
#> v purrr     0.3.4
#> -- Conflicts ------------------- tidymodels_conflicts() --
#> x purrr::discard() masks scales::discard()
#> x dplyr::filter()  masks stats::filter()
#> x dplyr::lag()     masks stats::lag()
#> x recipes::step()  masks stats::step()
library(modeltime.ensemble)
library(timetk)
library(tidyverse)
library(recipes)


df.total_pr1 <- read_csv("https://raw.githubusercontent.com/vmandela99/Training5_assignment_shiny/master/data.csv")
#> Parsed with column specification:
#> cols(
#>   Week = col_date(format = ""),
#>   Product_Name = col_character(),
#>   Demand_value = col_double()
#> )

df.total_pr1 %>%
  plot_seasonal_diagnostics(
    Week,Demand_value,
    .feature_set = c("week", "month.lbl"),
    .interactive = TRUE
  )
#> Warning: `group_by_()` is deprecated as of dplyr 0.7.0.
#> Please use `group_by()` instead.
#> See vignette('programming') for more help
#> This warning is displayed once every 8 hours.
#> Call `lifecycle::last_warnings()` to see where this warning was generated.
```

![](https://i.imgur.com/hRKSG8P.png)

``` r

splits <- splits <- df.total_pr1 %>%
  arrange(Week) %>% 
  time_series_split(date_var = Week,
                    assess = "40 weeks", cumulative = TRUE)

splits %>%
  tk_time_series_cv_plan() %>%
  plot_time_series_cv_plan(Week, Demand_value, .interactive = FALSE)
```

![](https://i.imgur.com/5PdIZus.png)

``` r

recipe_spec <- recipe(Demand_value ~ Week, df.total_pr1) %>%
  step_timeseries_signature(Week) %>%
  step_rm(matches("(iso$)|(xts$)|(day)|(hour)|(min)|(sec)|(am.pm)")) %>%
  step_mutate(Week_week = factor(Week_week, ordered = TRUE)) %>%
  step_dummy(all_nominal()) %>%
  step_normalize(contains("index.num"), Week_year)

recipe_spec %>% prep() %>% juice()
#> # A tibble: 52 x 35
#>    Week       Demand_value Week_index.num Week_year Week_half Week_quarter
#>    <date>            <dbl>          <dbl>     <dbl>     <int>        <int>
#>  1 2019-09-01        11773          0.792     0.550         2            3
#>  2 2019-10-01        13330          0.879     0.550         2            4
#>  3 2019-11-01         5743          0.969     0.550         2            4
#>  4 2019-12-01         6244          1.06      0.550         2            4
#>  5 2020-01-01         2560          1.15      1.57          1            1
#>  6 2020-03-01        24122          1.32      1.57          1            1
#>  7 2020-04-01         2980          1.41      1.57          1            2
#>  8 2020-05-01         1950          1.50      1.57          1            2
#>  9 2020-03-01          100          1.32      1.57          1            1
#> 10 2020-05-01          270          1.50      1.57          1            2
#> # ... with 42 more rows, and 29 more variables: Week_month <int>,
#> #   Week_mweek <int>, Week_week2 <int>, Week_week3 <int>, Week_week4 <int>,
#> #   Week_month.lbl_01 <dbl>, Week_month.lbl_02 <dbl>, Week_month.lbl_03 <dbl>,
#> #   Week_month.lbl_04 <dbl>, Week_month.lbl_05 <dbl>, Week_month.lbl_06 <dbl>,
#> #   Week_month.lbl_07 <dbl>, Week_month.lbl_08 <dbl>, Week_month.lbl_09 <dbl>,
#> #   Week_month.lbl_10 <dbl>, Week_month.lbl_11 <dbl>, Week_week_01 <dbl>,
#> #   Week_week_02 <dbl>, Week_week_03 <dbl>, Week_week_04 <dbl>,
#> #   Week_week_05 <dbl>, Week_week_06 <dbl>, Week_week_07 <dbl>,
#> #   Week_week_08 <dbl>, Week_week_09 <dbl>, Week_week_10 <dbl>,
#> #   Week_week_11 <dbl>, Week_week_12 <dbl>, Week_week_13 <dbl>

model_spec_glmnet <- linear_reg(penalty = 0.01, mixture = 0.5) %>%
  set_engine("glmnet")

#workflow
#elastic net
wflw_fit_glmnet <- workflow() %>%
  add_model(model_spec_glmnet) %>%
  add_recipe(recipe_spec %>% step_rm(Week)) %>%
  fit(training(splits))

#xgboost
model_spec_xgboost <- boost_tree() %>%
  set_engine("xgboost")

set.seed(123)
wflw_fit_xgboost <- workflow() %>%
  add_model(model_spec_xgboost) %>%
  add_recipe(recipe_spec %>% step_rm(Week)) %>%
  fit(training(splits))

#prophet
model_spec_prophet <- prophet_reg(
  seasonality_yearly = TRUE
) %>%
  set_engine("prophet") 

wflw_fit_prophet <- workflow() %>%
  add_model(model_spec_prophet) %>%
  add_recipe(recipe_spec) %>%
  fit(training(splits))
#> Disabling weekly seasonality. Run prophet with weekly.seasonality=TRUE to override this.
#> Disabling daily seasonality. Run prophet with daily.seasonality=TRUE to override this.

submodels_tbl <- modeltime_table(
  wflw_fit_glmnet,
  wflw_fit_xgboost,
  wflw_fit_prophet
)

submodels_tbl
#> # Modeltime Table
#> # A tibble: 3 x 3
#>   .model_id .model     .model_desc          
#>       <int> <list>     <chr>                
#> 1         1 <workflow> GLMNET               
#> 2         2 <workflow> XGBOOST              
#> 3         3 <workflow> PROPHET W/ REGRESSORS

submodels_tbl %>% 
  modeltime_accuracy(testing(splits)) %>%
  table_modeltime_accuracy(.interactive = FALSE)
#> Warning: Problem with `mutate()` input `.nested.col`.
#> i There are new levels in a factor: 14, 27
#> i Input `.nested.col` is `purrr::map(...)`.
#> Warning: There are new levels in a factor: 14, 27
#> Warning: Problem with `mutate()` input `.nested.col`.
#> i There are new levels in a factor: 14, 27
#> i Input `.nested.col` is `purrr::map(...)`.
#> Warning: There are new levels in a factor: 14, 27
#> Warning: Problem with `mutate()` input `.nested.col`.
#> i There are new levels in a factor: 14, 27
#> i Input `.nested.col` is `purrr::map(...)`.
#> Warning: There are new levels in a factor: 14, 27
#> Warning: Problem with `mutate()` input `.nested.col`.
#> i Values are not uniquely identified; output will contain list-cols.
#> * Use `values_fn = list` to suppress this warning.
#> * Use `values_fn = length` to identify where the duplicates arise
#> * Use `values_fn = {summary_fun}` to summarise duplicates
#> i Input `.nested.col` is `purrr::map(...)`.
#> Warning: Values are not uniquely identified; output will contain list-cols.
#> * Use `values_fn = list` to suppress this warning.
#> * Use `values_fn = length` to identify where the duplicates arise
#> * Use `values_fn = {summary_fun}` to summarise duplicates
#> Warning: Problem with `mutate()` input `.nested.col`.
#> i There are new levels in a factor: 14, 27
#> i Input `.nested.col` is `purrr::map(...)`.
#> Warning: There are new levels in a factor: 14, 27
#> Warning: Problem with `mutate()` input `.nested.col`.
#> i There are new levels in a factor: 14, 27
#> i Input `.nested.col` is `purrr::map(...)`.
#> Warning: There are new levels in a factor: 14, 27
#> Warning: Problem with `mutate()` input `.nested.col`.
#> i There are new levels in a factor: 14, 27
#> i Input `.nested.col` is `purrr::map(...)`.
#> Warning: There are new levels in a factor: 14, 27
#> Warning: Problem with `mutate()` input `.nested.col`.
#> i Values are not uniquely identified; output will contain list-cols.
#> * Use `values_fn = list` to suppress this warning.
#> * Use `values_fn = length` to identify where the duplicates arise
#> * Use `values_fn = {summary_fun}` to summarise duplicates
#> i Input `.nested.col` is `purrr::map(...)`.
#> Warning: Values are not uniquely identified; output will contain list-cols.
#> * Use `values_fn = list` to suppress this warning.
#> * Use `values_fn = length` to identify where the duplicates arise
#> * Use `values_fn = {summary_fun}` to summarise duplicates
#> Warning: Problem with `mutate()` input `.nested.col`.
#> i There are new levels in a factor: 14, 27
#> i Input `.nested.col` is `purrr::map(...)`.
#> Warning: There are new levels in a factor: 14, 27
#> Warning: Problem with `mutate()` input `.nested.col`.
#> i There are new levels in a factor: 14, 27
#> i Input `.nested.col` is `purrr::map(...)`.
#> Warning: There are new levels in a factor: 14, 27
#> Error: Problem with `mutate()` input `.is_null`.
#> x object '.calibration_data' not found
#> i Input `.is_null` is `purrr::map_lgl(.calibration_data, is.null)`.
```

<sup>Created on 2020-10-19 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by mdancho84** [comment]
The error indicates that you have new factor levels 14 and 27 that are present in your test set. These categories were not present in your training set. You'll need to rectify this.

**Comment by vmandela99** [comment]
Thanks big time for the superfast response matt.

**Comment by mdancho84** [comment]
You got it - Anything modeltime / timetk - I'm trying to iron out as much as possible as fast as possible. 

Thanks for your help. Appreciate the GH issue because I've been seeing this from more than one user.
